src/app/recover-funds/[address]/RecoverFundsButton.tsx

```diff
@@ -143,26 +157,47 @@ function RecoverFundsButton({
 
   if (!signer) return null;
 
-  if (chain?.id !== 1 && chain?.id !== 5) {
+  if (
+    chain?.id !== mainnet.id &&
+    chain?.id !== goerli.id &&
+    chain?.id !== sepolia.id
+  ) {
     return (
       <div>Unknown L1 chain id. This chain is not supported by this tool</div>
     );
   }
 
-  if (chain?.id !== chainID) {
+  if (getTargetChainId(chain?.id) !== chainID) {
     return (
       <div>
-        To recover funds, connect to chain ${chain?.id} (${chain?.name})
+        To recover funds, connect to chain {chain?.id} ({chain?.name})
       </div>
     );
   }
 
   return (
     <>
       <div className="recover-funds-form">
-        <button id="recover-button" disabled={loading} onClick={handleRecover}>
-          Recover
-        </button>
+        <div>
+          <button
+            className="button"
+            id="recover-button"
+            disabled={loading}
+            onClick={handleRecover}
+          >
+            Recover
+          </button>
+        </div>
+        {loading && isSmartContractWallet && (
+          <div className="flex flex-col">
+            <span>
+              <b>
+                To continue, please approve tx on your smart contract wallet.
+              </b>{' '}
+              If you have k of n signers, then k of n will need to sign.
+            </span>
+          </div>
```

Comment:
nitpick: text appear aligned to the left, not centered like the rest of the texts